string message
string display_type
